packages/expo-av/src/Audio/Recording.ts

```diff
@@ -121,6 +125,10 @@ export const RECORDING_OPTIONS_PRESET_HIGH_QUALITY: RecordingOptions = {
     linearPCMIsBigEndian: false,
     linearPCMIsFloat: false,
   },
+  web: {
+    mimeType: 'audio/webm',
```

Comment:
I wonder if anyone has any input on what do to about this default mimeType, audio/webm will work in Chrome and I think Firefox.

Should we leave it at audio/webm which I think will work for Chrome/Firefox?
Document that the developer will have to use MediaRecorder.isTypeSupported to find a supported mimeType?
Have no default and require the developer finds a supported mimeType?

I'm currently experimenting with the following to prefer AAC (iOS 14 Safari seems to be able to do this) with several fallbacks (opus, vp8, polyfill mp3, etc.)
These changes aren't committed yet, I'm still checking them especially on iOS.
I also found that at least on iOS Safari `bitsPerSecond` is ignored and I have to set `audioBitsPerSecond` to have any effect, I also have `extension` to I know if the files require backend transcoding.

```TypeScript
const recordingSettings = React.useMemo(
  () => ({
    android: {
      extension: '.m4a',
      outputFormat: Audio.RECORDING_OPTION_ANDROID_OUTPUT_FORMAT_MPEG_4,
      audioEncoder: Audio.RECORDING_OPTION_ANDROID_AUDIO_ENCODER_HE_AAC,
      sampleRate: 44100,
      numberOfChannels: 2,
      bitRate: 64000,
    },
    ios: {
      extension: '.m4a',
      outputFormat: Audio.RECORDING_OPTION_IOS_OUTPUT_FORMAT_MPEG4AAC_HE,
      audioQuality: Audio.RECORDING_OPTION_IOS_AUDIO_QUALITY_MIN,
      sampleRate: 44100,
      numberOfChannels: 2,
      bitRate: 64000,
      linearPCMBitDepth: 16,
      linearPCMIsBigEndian: false,
      linearPCMIsFloat: false,
    },
    web: (() => {
      const mimes = [
        {
          mimeType: 'audio/mp4;codecs="mp4a.40.5"', // MPEG-4 HE-AAC v1
          audioBitsPerSecond: 64000,
          bitsPerSecond: 64000,
          extension: '.m4a',
        },
        {
          mimeType: 'audio/mp4;codecs="mp4a.40.2"', // MPEG-4 AAC LC
          audioBitsPerSecond: 64000,
          bitsPerSecond: 64000,
          extension: '.m4a',
        },
        {
          mimeType: 'audio/webm;codecs="opus"',
          audioBitsPerSecond: 64000,
          bitsPerSecond: 64000,
          extension: '.webm',
        },
        {
          mimeType: 'audio/webm;codecs="vp8"',
          audioBitsPerSecond: 64000,
          bitsPerSecond: 64000,
          extension: '.webm',
        },
        {
          mimeType: 'audio/webm',
          audioBitsPerSecond: 64000,
          bitsPerSecond: 64000,
          extension: '.webm',
        },
        {
          mimeType: 'audio/mpeg', // Support depends on polyfill
          audioBitsPerSecond: 128000,
          bitsPerSecond: 128000,
          extension: '.mp3',
        },
      ]

      for (let index = 0; index < mimes.length; index++) {
        const mime = mimes[index]

        if ((window as any).MediaRecorder.isTypeSupported(mime.mimeType)) {
          return mime
        }
      }

      return false
    })(),
  }),
  [window.MediaRecorder],
)
```

Reply:
As this stands, the web code would run on Android & iOS where `window.MediaRecorder` is not available?

Apologies if I have misinterpreted.

As an approach though, this looks fine?

Reply:
This was really just showing how it'd need to be used by a client app if you wanted to pick different mimeTypes/codecs depending upon what the browser supports.

You're right though the web code should be wrapped in a `Platform.OS === 'web'` condition (I returned undefined in my app) and the window.MediaRecorder shouldn't be necessary depending on if/how you are polyfilling.

I'm not sure if there any/much benefit using useMemo, if I recall its because you might need it within your render if the MediaRecorder is polyfilled, I doubt there's much cost in calling isTypeSupported so it's likely unnecessary.

In my app I haven't bothered using the useMemo.